### PR TITLE
Printing tests for HybridFactorGraph and DCGaussianMixtureFactor

### PR DESCRIPTION
**Pull request by varunagrawal**
I added a bunch of tests for printing (since I spent that time adding `assert_print_equal` years ago) in an effort to simplify the printing logic for the factors involving `DecisionTree`s.

Indentation is already present so it makes sense to remove the regex stuff.

### CODE REVIEW

**Review by dellaert — APPROVED** [review]
Seems not a priority. Focus.

**Comment by varunagrawal** [comment]
Did this since it was a point of review in a previous PR and I'm waiting on the CI for GaussianHybridFactorGraph.

**Comment by dellaert** [comment]
> Did this since it was a point of review in a previous PR and I'm waiting on the CI for GaussianHybridFactorGraph.

Must mean paper is written and Anymal results are awesome, including slip detection

File: gtsam/hybrid/tests/testDCGaussianMixtureFactor.cpp [hunk 1]
```diff
@@ -88,6 +89,51 @@ TEST(DCGaussianMixtureFactor, Sum) {
...
+  DCGaussianMixtureFactor mixtureFactor({X(1), X(2)}, {m1}, factors);
+
+  std::string expected = \
+R"([ x1 x2; 1 ]{
+ Choice(1) 
+ 0 Leaf Jacobian factor on 2 keys: 
+  A[x1] = [
+	0;
+	0
+]
```
Comment (ProfFan) [inline gtsam/hybrid/tests/testDCGaussianMixtureFactor.cpp:110]: this is not indented

**Review by ProfFan — COMMENTED** [review]
Let's postpone this discussion and finish RSS first?

File: gtsam/hybrid/tests/testDCGaussianMixtureFactor.cpp (hunk 1 above)
Comment (varunagrawal) [reply gtsam/hybrid/tests/testDCGaussianMixtureFactor.cpp:110]: Do we need the indentation though? The output is pretty straightforward to parse and it has sufficient indentation.

But yeah, we can park this for later.

Comment (dellaert) [reply gtsam/hybrid/tests/testDCGaussianMixtureFactor.cpp:110]: Well, I'd merge it now, work has been done. Just don't dwell on it.

**Pull request merged**